GVRf/Framework/jni/shaders/material/unlit_shader.cpp

```diff
@@ -85,6 +85,11 @@ void UnlitShader::render(const glm::mat4& mvp_matrix,
         throw error;
     }
 
+    if (texture->getId() == 0) {
+        std::string error = "UnlitShader::render : texture with invalid Id";
```

Comment:
Based on my experience as a user of GVRF, it is helpful to be able to get diagnostics about (incorrect) usage of recycled objects. It could direct the application developer's attention to the right thing. I know something like this would have saved me time - I had the same problem as the one in VideoScript.java.

And to answer your question: Texture::getId() proxies to GLTexture::id(); so looking at GLTexture, glGenTextures is not going to write a texture name of 0, after assignment id_ is not mutated; if glGenTextures fails, then id_ could be anything; including 0, I guess.


---
Should this "if getId() == 0" check be applied to other renderers?